### ISSUE TITLE
Document the purpose of globalScript

### ISSUE THREAD

**Issue by simonhaenisch** [ionic-team/stencil-site#374]
One of the few things I'm still not entirely sure about is the purpose of the `globalScript`... at the moment mine looks like

```ts
import '@ionic/pwa-elements';
import { setupConfig } from '@ionic/core';

setupConfig({ mode: 'md' });
```

but I'm not sure why I should do this stuff in here and not e. g. in my root component file (`app-root.tsx`)?

Things that I think would be helpful to get documented:

* Purpose of the global script, i. e. when do I need it?
* Best practices about what should go in there, i. e. what do I put in?

If someone could answer me those questions (/cc @manucorporat or @adamdbradley), I'm happy to make a PR to update https://stenciljs.com/docs/config.

**Comment by bitflower** [comment]
I imagine one reason being to do stuff that's not "presentational" in the sense that component "present" things. While you would configure things here in this global script.

**Comment by claviska** [comment]
This was fixed in #479. Here's the relevant section in the docs: https://stenciljs.com/docs/config#globalscript
